### PR TITLE
Send highlighting to SonarQube

### LINKED ISSUE #55: Support source code highlighting

### PR DESCRIPTION
**Pull request by alban-auzeill**
Fix #55

### CODE REVIEW

File: common-rule-engine/src/main/java/org/sonar/uast/UastNode.java [hunk 1]
```diff
@@ -83,14 +83,18 @@ public String toString() {
     CASE,
     CLASS,
     COMMENT,
+    STRUCTURED, // COMMENT specifier
```
Comment (SonarTech) [inline common-rule-engine/src/main/java/org/sonar/uast/UastNode.java]: Code Smell Code Smell: Move this trailing comment on the previous empty line. ([squid:TrailingCommentCheck](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ATrailingCommentCheck))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.go%3Asonar-go&pullRequest=119&issues=AWIlzOvxH1UYAK0FSe8f&open=AWIlzOvxH1UYAK0FSe8f)

[ISSUE_KEY]: AWIlzOvxH1UYAK0FSe8f

File: common-rule-engine/src/main/java/org/sonar/uast/UastNode.java [hunk 2]
```diff
@@ -83,14 +83,18 @@ public String toString() {
...
     CONDITION,
     ELSE,
     EOF,
     FUNCTION,
     IDENTIFIER,
+    TYPE,
     IF,
+    KEYWORD,
     LITERAL,
+    STRING, // LITERAL specifier
```
Comment (SonarTech) [inline common-rule-engine/src/main/java/org/sonar/uast/UastNode.java]: Code Smell Code Smell: Move this trailing comment on the previous empty line. ([squid:TrailingCommentCheck](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ATrailingCommentCheck))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.go%3Asonar-go&pullRequest=119&issues=AWIlzOvxH1UYAK0FSe8g&open=AWIlzOvxH1UYAK0FSe8g)

[ISSUE_KEY]: AWIlzOvxH1UYAK0FSe8g

File: sonar-go-plugin/src/main/java/org/sonar/go/plugin/GoSensor.java [hunk 3]
```diff
@@ -79,6 +80,12 @@ private void saveMetrics(Metrics metrics, SensorContext context, InputFile input
     fileLinesContext.save();
   }
 
+  private void saveHighlighting(UastNode uast, SensorContext context, InputFile inputFile) {
```
Comment (SonarTech) [inline sonar-go-plugin/src/main/java/org/sonar/go/plugin/GoSensor.java]: Code Smell Code Smell: Make "saveHighlighting" a "static" method. ([squid:S2325](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.go%3Asonar-go&pullRequest=119&issues=AWIlzOwNH1UYAK0FSe8h&open=AWIlzOwNH1UYAK0FSe8h)

[ISSUE_KEY]: AWIlzOwNH1UYAK0FSe8h

File: sonar-go-plugin/src/main/java/org/sonar/go/plugin/HighlightingVisitor.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  public HighlightingVisitor(SensorContext sensorContext, InputFile inputFile) {
+    this.inputFile = inputFile;
+    newHighlighting = sensorContext.newHighlighting()
+      .onFile(inputFile);
+  }
+
+  public void visit(UastNode node) {
+    UastNode.Token token = node.token;
+    if (token != null) {
+      Set<Kind> kinds = node.kinds;
```
Comment (SonarTech) [inline sonar-go-plugin/src/main/java/org/sonar/go/plugin/HighlightingVisitor.java]: Code Smell Code Smell: Convert this Set to an EnumSet. ([squid:S1641](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1641))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.go%3Asonar-go&pullRequest=119&issues=AWIlzOwSH1UYAK0FSe8i&open=AWIlzOwSH1UYAK0FSe8i)

[ISSUE_KEY]: AWIlzOwSH1UYAK0FSe8i

**Review by saberduck — APPROVED** [review]
please fix the conflicts, otherwise LGTM . We can discuss renaming when we migrate to protobuf

**Pull request merged**